src/Symfony/Component/AssetMapper/Tests/ImportMap/Resolver/JsDelivrEsmResolverTest.php

```diff
@@ -235,21 +235,49 @@ public static function provideResolvePackagesTests(): iterable
         ];
     }
 
-    public function testImportRegex()
+    /**
+     * @dataProvider getRegexTests
+     */
+    public function testImportRegex(string $subject, array $expectedPackages)
     {
-        $subject = 'import{Color as t}from"/npm/@kurkle/color@0.3.2/+esm";import t from"/npm/jquery@3.7.0/+esm";import e from"/npm/popper.js@1.16.1/+esm";console.log("yo");';
         preg_match_all(JsDelivrEsmResolver::IMPORT_REGEX, $subject, $matches);
 
-        $this->assertCount(3, $matches[0]);
-        $this->assertSame([
-            '@kurkle/color',
-            'jquery',
-            'popper.js',
-        ], $matches[1]);
-        $this->assertSame([
-            '0.3.2',
-            '3.7.0',
-            '1.16.1',
-        ], $matches[2]);
+        $this->assertCount(\count($expectedPackages), $matches[0]);
+        $expectedNames = [];
+        $expectedVersions = [];
+        foreach ($expectedPackages as $packageData) {
+            $expectedNames[] = $packageData[0];
+            $expectedVersions[] = $packageData[1];
+        }
+        $this->assertSame($expectedNames, $matches[1]);
+        $this->assertSame($expectedVersions, $matches[2]);
+    }
+
+    public function getRegexTests()
```

Comment:
```suggestion
    public static function getRegexTests()
```

Reply:
I don't know if I'll EVER remember that 🤦  - thanks